File: movaro/apps/app/lib/features/city_insights/domain/entities/city_insight_explore_place_entity.dart
import 'package:movaro_app/features/city_insights/domain/entities/city_insight_entity.dart';

class CityInsightExplorePlaceEntity {
  const CityInsightExplorePlaceEntity({
    required this.id,
    required this.cityId,
    required this.cityName,
    required this.theme,
    required this.name,
    required this.category,
    required this.shortText,
    required this.mapUrl,
    required this.source,
    this.neighborhood,
    this.region,
    this.imageUrl,
    this.latitude,
    this.longitude,
  });

  final String id;
  final String cityId;
  final String cityName;
  final CityInsightTheme theme;
  final String name;
  final String category;
  final String? neighborhood;
  final String? region;
  final String shortText;
  final String? imageUrl;
  final String mapUrl;
  final double? latitude;
  final double? longitude;
  final String source;
}

File: movaro/apps/api/src/modules/city-insights/domain/entities/city-insight-explore-place.entity.ts
import { CityInsightTheme } from './city-insight.entity';

export interface CityInsightExplorePlaceEntity {
  id: string;
  cityId: string;
  cityName: string;
  theme: CityInsightTheme;
  name: string;
  category: string;
  neighborhood?: string;
  region?: string;
  shortText: string;
  imageUrl?: string;
  mapUrl: string;
  latitude?: number;
  longitude?: number;
  source: 'openstreetmap' | 'template' | 'curated';
}
